### PR TITLE
Remove some IRouter methods

### PR DESCRIPTION
**Pull request by rullzer**
This is not the end. IRouter needs to burn.
But it is a start.

🎵 we didn't start the fire 🎵

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]
:fire: :koala: :broom:

File: lib/public/Route/IRouter.php [hunk 1]
```diff
@@ -72,15 +47,6 @@ public function loadRoutes($app = null);
...
 	public function useCollection($name);
 
-	/**
-	 * returns the current collection name in use for adding routes
-	 *
-	 * @return string the collection name
-	 * @since 8.0.0
-	 * @deprecated 9.0.0
-	 */
-	public function getCurrentCollection();
```
Comment (MorrisJobke) [inline lib/public/Route/IRouter.php:82]: This is used in https://github.com/nextcloud/server/blob/cb057829f72c70e819f456edfadbb29d72dba832/lib/private/AppFramework/Routing/RouteConfig.php#L93

As the `$this->router` is type hinted as `IRouter` it needs to be changed to `Router` first.

Comment (rullzer) [reply lib/public/Route/IRouter.php:81]: Did some more cleanup should be good to go now

**Review by ChristophWurst — APPROVED** [review]
Still :+1: from me

**Review by juliusknorr — APPROVED** [review]
:+1:

**Comment by juliushaertl** [comment]
Rebased and fixed the php cs issue

**Comment by MorrisJobke** [comment]
Added to #23210

**Pull request merged**